src/components/dropdown/daikin-dropdown.ts

```diff
@@ -0,0 +1,362 @@
+
+  private _startAutoUpdate() {
+    if (!isClient) {
+      return;
+    }
+
+    const button = this._buttonRef.value;
+    const contents = this._contentsRef.value;
+    if (!button || !contents) {
+      return;
+    }
+
+    // TODO(DDS-1226): refactor here with Popover API + CSS Anchor Positioning instead of using floating-ui
+    this._autoUpdateCleanup?.();
+    this._autoUpdateCleanup = autoUpdate(button, contents, () => {
+      computePosition(button, contents, {
+        placement: "bottom",
+        middleware: [
+          flip({ fallbackStrategy: "initialPlacement" }),
+          offset({ mainAxis: -1 }),
+        ],
+      })
+        .then(({ x, y }) => {
+          Object.assign(contents.style, {
+            left: `${x}px`,
+            top: `${y}px`,
+          });
+        })
+        .catch((e: unknown) => console.error(e));
+    });
+  }
+
+  private _uninstallAutoUpdate() {
+    this.open = false;
+    this._autoUpdateCleanup?.();
+    this._autoUpdateCleanup = null;
+  }
+
+  private _reflectButtonLabel() {
+    const items = this._items;
+
+    const itemIndex: number = items.findIndex(
+      ({ value }) => this.value === value
+    );
+
+    if (itemIndex > -1) {
+      for (const item of items) {
+        item.selected = false;
+      }
+      this._buttonLabel = items[itemIndex].textContent ?? "";
+      this._buttonLabelType = "value";
+      items[itemIndex].selected = true;
+    } else {
+      this._buttonLabel = this.placeholder;
+    }
+  }
+
+  private _handleClick() {
+    this.open = !this.open;
+  }
+
+  private _handleKeyDown(event: KeyboardEvent): void {
+    const moveOffset = {
+      ArrowDown: 1,
+      ArrowUp: -1,
+    }[event.key];
+    if (!moveOffset) {
+      return;
+    }
+
+    const items = this._items;
+
+    // Get focused item if any
+    const activeElement = document.activeElement;
+
+    const focusedItemIndex = activeElement
+      ? items.findIndex((item) => item.contains(activeElement))
+      : -1;
+
+    // If there is no item focused, focus on first item
+    if (focusedItemIndex < 0) {
+      this._items[0].focus();
+      event.preventDefault();
+      return;
+    }
+
+    // If there is a item focused, move focus forward or backward
+    for (let i = 1; i <= items.length; i++) {
+      const index =
+        (focusedItemIndex + moveOffset * i + items.length * i) % items.length;
+      const candidate = items[index];
+
+      candidate.focus();
+      event.preventDefault();
+      return;
+    }
+  }
+
+  /**
+   * Handle `select` event from `daikin-dropdown-item`.
+   */
+  private _handleSelect(e: SelectEvent) {
+    this.value = e.detail.value;
+
+    const item = this._items.find((item) => item.value === this.value);
+    if (item) {
+      this._buttonLabel = item.textContent ?? "";
+    }
+
+    const event = new CustomEvent("change", {
+      detail: {
+        value: this.value,
+      },
+    });
+    this.open = false;
+
+    this.dispatchEvent(event);
+  }
```

Comment:
We should not react to "select" event dispatched from unmanaged elements.
Also, we don't have to add `detail` as the user can retrieve the current value via `event.target.value`.

```suggestion
  private _handleSelect(event: SelectEvent) {
    const item = event.target;
    if (!this._items.include(item)) {
      return;
    }

    this._hasSelectedItem = true;
    this._selectedItemLabel = event.target.textContent ?? "";

    this.open = false;
    this.value = e.detail.value;

    this.dispatchEvent(new Event("change"));
  }
```